js-api-spec/value/color.test.ts

```diff
@@ -26,22 +49,1008 @@ function hsl(
+    channels: ['x', 'y', 'z'] as ChannelNameXyz[],
+    ranges: [
+      [0, 1],
+      [0, 1],
+      [0, 1],
+    ],
+    hasOutOfGamut: false,
+    gamutExamples: [
+      [
+        [0.5, 2, 2],
+        [0.5, 2, 2],
+      ],
+    ],
+  },
+  xyzD50: {
+    constructor: xyzD50,
+    name: 'xyz-d50',
+    isLegacy: false,
+    isPolar: false,
+    pink: [0.6640698533004002, 0.5367266625281085, 0.4345958246720296],
+    blue: [0.08408207011375313, 0.10634498228480066, 0.1470370877550857],
+    channels: ['x', 'y', 'z'] as ChannelNameXyz[],
+    ranges: [
+      [0, 1],
+      [0, 1],
+      [0, 1],
+    ],
+    hasOutOfGamut: false,
+    gamutExamples: [
+      [
+        [0.5, 2, 2],
+        [0.5, 2, 2],
+      ],
+    ],
+  },
+  xyzD65: {
+    constructor: xyzD65,
+    name: 'xyz',
+    isLegacy: false,
+    isPolar: false,
+    pink: [0.6495957411726918, 0.5323965129525022, 0.575341840710865],
+    blue: [0.08718323686632441, 0.1081164314257634, 0.19446762910683627],
+    channels: ['x', 'y', 'z'] as ChannelNameXyz[],
+    ranges: [
+      [0, 1],
+      [0, 1],
+      [0, 1],
+    ],
+    hasOutOfGamut: false,
+    gamutExamples: [
+      [
+        [0.5, 2, 2],
+        [0.5, 2, 2],
+      ],
+    ],
+  },
+  rgb: {
+    constructor: rgb,
+    name: 'rgb',
+    isLegacy: true,
+    isPolar: false,
+    pink: [254.9945293093454, 168.28594090628783, 193.38511936687908],
+    blue: [38.144364133784602, 100.003690461188378, 120.626489290564506],
+    channels: ['red', 'green', 'blue'] as ChannelNameRgb[],
+    ranges: [
+      [0, 255],
+      [0, 255],
+      [0, 255],
+    ],
+    hasOutOfGamut: true,
+    gamutExamples: [
+      [
+        [300, 300, 300],
+        [255, 255, 255],
+      ],
+    ],
+  },
+  hsl: {
+    constructor: hsl,
+    name: 'hsl',
+    isLegacy: true,
+    isPolar: true,
+    pink: [342.63204677447646, 99.98738302509669, 82.99617063051632],
+    blue: [195.0016494775154, 51.95041997811069, 31.009932309443183],
+    channels: ['hue', 'saturation', 'lightness'] as ChannelNameHsl[],
+    hasPowerless: true,
+    ranges: [
+      [0, 360],
+      [0, 100],
+      [0, 100],
+    ],
+    hasOutOfGamut: false,
+    gamutExamples: [
+      [
+        [0.5, 100, 50],
+        [0.5, 100, 50],
+      ],
+    ],
+  },
+  hwb: {
+    constructor: hwb,
+    name: 'hwb',
+    isLegacy: true,
+    isPolar: true,
+    pink: [342.63204677447646, 65.99448662991679, 0.002145368884157506],
+    blue: [195.0016494775154, 14.900142239759612, 52.880277620873244],
+    channels: ['hue', 'whiteness', 'blackness'] as ChannelNameHwb[],
+    hasPowerless: true,
+    ranges: [
+      [0, 360],
+      [0, 100],
+      [0, 100],
+    ],
+    hasOutOfGamut: false,
+    gamutExamples: [
+      [
+        [0.5, 100, 50],
+        [0.5, 100, 50],
+      ],
+    ],
+  },
+};
+
+type InterpolationExample = [
+  {weight: number; method?: string},
+  [number, number, number]
+];
+
+const interpolations: {[space: string]: InterpolationExample[]} = {
```

Comment:
Where are you sourcing these examples? If they're from a w3c source, we should link it.

Reply:
I generated them using Color.js, with the code living in a [branch in Oddbird's fork](https://github.com/oddbird/color.js/blob/sass-interpolations/apps/mixes/mixes.js) of Color.js due to some internal changes to Color.js I needed to make at the time to support the Hue interpolation method. 

I didn't write the generation code planning for it to be incorporated into sass-spec, but I'd be open to that if you have ideas on how to do that.

Reply:
I meant more, what caused you to choose these specific colors to test against rather than any others?

Reply:
For starting colors, I picked 2 of Oddbird's brand colors, not for any particular reason about the colors themselves. The interpolation examples are basically between those colors with weights [0, 0.5, 1], and then for polar spaces, each hue interpolation method. My thought was to verify a range of options, as opposed to checking specific edge cases, but the checks could easily be expanded.